src/Sulu/Bundle/AdminBundle/Resources/js/services/ContentLocaleUpdater/updateRouterAttributesFromUserStoreContentLocale.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+import type {UpdateAttributesHook} from '../Router/types';
+import pathToRegexp from 'path-to-regexp';
+import {toJS} from 'mobx';
+import userStore from '../../stores/userStore/userStore';
+
+const updateRouterAttributesFromUserStoreContentLocale: UpdateAttributesHook = function(route, attributes) {
+    // do nothing when locale is explicit set
+    if (attributes.locale) {
+        return attributes;
+    }
+
+    const keys = [];
+    pathToRegexp(route.path, keys);
+    const keyNames = keys.map((key) => key.name);
+
+    // do nothing when the route does not require a locale
+    if (!keyNames.includes('locale')) {
+        return attributes;
+    }
+
+    const locales = toJS(route.options.locales);
+
+    // set content locale if route accept the current content locale
+    if (locales && locales.includes(userStore.contentLocale)) {
```

Comment:
As I've already said I don't like that a lot of assumptions are made here (there has to be a `locale` and `locales` parameters), and it conflicts when the these parameters are used in other views in a different way, resulting in weird behavior.